### DISCUSSION TITLE
NVIDIA Container Runtime and hardware-specific libraries

### DISCUSSION THREAD

**Discussion by maggu2810** [OE4T/meta-tegra#1237] [General]
In the Wiki you (@madisongh) are writing

> The [Docker containers that NVIDIA supplies](https://ngc.nvidia.com/catalog/containers?orderBy=modifiedDESC&pageNumber=0&query=jetson&quickFilter=containers&filters=) do not bundle in most of the hardware-specific libraries needed to run them, but expect them to be provided by the underlying host OS, so be sure to include TensorRT ([note](https://github.com/madisongh/meta-tegra/wiki/L4T-R32.3.1-Notes#tensorrt-packaging-change)), CuDNN, and/or VisionWorks, if you expect to be running containers needing those packages.

Can you tell me if it is possible (and how) to use a different L4T host system version and L4T container version?

For example use "L4T R35.3.1/JetPack 5.1.1" host system and "L4T R32.6.1/JetPack 4.6" container. Could this work after adding the hardware specific libraries into the container (in the container specific version)?

**Comment by dwalkes** [comment]
> Can you tell me if it is possible (and how) to use a different L4T host system version and L4T container version?
> For example use "L4T R35.3.1/JetPack 5.1.1" host system and "L4T R32.6.1/JetPack 4.6" container

For Jetpack 4.6 OS and Jetpack 5.x base I don't think this will be possible, and even if it was possible I would suspect you might find hard or impossible to debug situations related to closed source libraries.

NVIDIA is moving the direction of decoupling their containers from the rootfs, see the notes at https://catalog.ngc.nvidia.com/orgs/nvidia/containers/l4t-base regarding the changes starting with Jetpack 5.0.  This won't be exactly what you are asking for here but should provide some flexibility which wasn't there in Jetpack 4.

I personally wouldn't recommend using a production combination NVIDIA doesn't specifically support, given the amount of closed source libraries involved.
